packages/api-key-stamper/src/tink/elliptic_curves.ts

```diff
@@ -136,35 +136,86 @@ function getY(x: bigint, lsb: boolean): bigint {
 }
 
 /**
- * Decodes a public key in _compressed_ format.
+ *
+ * Given x and y coordinates of a JWK, checks whether these are valid points on
+ * the P-256 elliptic curve.
+ *
+ * P-256 only
+ *
+ * @param x x-coordinate
+ * @param y y-coordinate
+ * @return boolean validitiy
+ */
+function validateUncompressedXY(x: bigint, y: bigint): boolean {
+  const p = getModulus();
+  const a = p - BigInt(3);
+  const b = getB();
+  const rhs = ((x * x + a) * x + b) % p;
+  const lhs = y ** BigInt(2) % p;
+  return lhs === rhs;
+}
+
+/**
+ * Decodes a public key in _compressed_ OR _uncompressed_ format.
  * Augmented to ensure that the x and y components are padded to fit 32 bytes.
  *
  * P-256 only
  */
 export function pointDecode(point: Uint8Array): JsonWebKey {
   const fieldSize = fieldSizeInBytes();
-
-  if (point.length !== 1 + fieldSize) {
-    throw new Error("compressed point has wrong length");
-  }
-  if (point[0] !== 2 && point[0] !== 3) {
-    throw new Error("invalid format");
+  const compressedLength = fieldSize + 1;
+  const uncompressedLength = 2 * fieldSize + 1;
+  if (
+    point.length !== compressedLength &&
+    point.length !== uncompressedLength
+  ) {
+    throw new Error(
+      "Invalid length: point is not in compressed or uncompressed format"
+    );
   }
-  const lsb = point[0] === 3; // point[0] must be 2 (false) or 3 (true).
-  const x = byteArrayToInteger(point.subarray(1, point.length));
-  const p = getModulus();
-  if (x < BigInt(0) || x >= p) {
-    throw new Error("x is out of range");
+  // Decodes point if its length and first bit match the compressed format
+  if ((point[0] === 2 || point[0] === 3) && point.length == compressedLength) {
```

Comment:
We can draw inspiration from the original implementation, which we're getting closer to given our support for multiple point types (compressed + uncompressed): https://github.com/tink-crypto/tink/blob/6f74b99a2bfe6677e3670799116a57268fd067fa/javascript/subtle/elliptic_curves.ts#L586-L652

We can borrow some of their implementation for checks, like
```
if (point.length !== 1 + 2 * fieldSize || point[0] !== 4) {
        throw new InvalidArgumentsException('invalid point');
      }
```

Reply:
hmmm it looks like they have an argument called `pointFormatType` which can be something like `UNCOMPRESSED` or `COMPRESSED` and then within the function body they check whether given a point format type, the input matches the format. 

Do we want to mimic this entirely? 

Right now I'm just doing a check to match compressed format on 177, in which I decode compressed. 

```
// Decodes point if its length and first bit match the compressed format
  if ((point[0] === 2 || point[0] === 3) && point.length == compressedLength) {
```

And if it doesn't match 177 then on 194 I check for matching uncompressed (effectively the inverse of what they check for). 

```
    // Decodes point if its length and first bit match the uncompressed format
  } else if (point[0] === 4 && point.length == uncompressedLength) {
```

Just to clarify -- are you saying we should move towards passing in a parameter for point type???